### ISSUE TITLE
c-writer fails on un-exited loop with value

### ISSUE THREAD

**Issue by kripken** [WebAssembly/wabt#674]
Testcase:
````
(module
 (func $0 (result i32)
  (loop $label$1 (result i32)
   (br $label$1)
  )
 )
)
````
Hits this error:
````
wasm2c: /wabt/src/c-writer.cc:712: void wabt::{anonymous}::CWriter::Write(const wabt::{anonymous}::StackVar&): Assertion `index < type_stack_.size()' failed.
````

**Comment by binji** [comment]
Thanks for taking a look! Lemme see...

**Comment by binji** [comment]
Oh, right. This is that bug that I write every time where br to loop is special. :-|

**Comment by binji** [comment]
Should be fixed in d25829e02ca6ff1905d76534e5f06a56d8f84e8c.
